.github/workflows/test_action_locally.yml

```
@@ -0,0 +1,162 @@
+name: Test the action
+on: 
+  push:
```

I think we can define the scope this way, but maybe we can leave the `pull_request` open to all scenarios and not only against `main` branch

```suggestion
on:
  pull_request:
    branches:
    - main 
  push:
    branches:
    - main
```